filament/src/Material.cpp

```diff
@@ -313,6 +311,15 @@ FMaterial::FMaterial(FEngine& engine, const Material::Builder& builder)
 }
 
 FMaterial::~FMaterial() noexcept {
+
+#if FILAMENT_ENABLE_MATDBG
+    // Unregister the material with matdbg.
+    matdbg::DebugServer* server = upcast(mEngine).debug.server;
+    if (UTILS_UNLIKELY(server)) {
+        server->removeMaterial(mDebuggerId);
+    }
+#endif
+
```

Comment:
it would be better to move this to `terminate()`. Anything that could fail shouldn't be in a dtor.

Reply:
Done.